Option to use USGS geometry

pjhartzell opened the pull request:
**Related Issue(s):**
None

**Description:**
Added ability to pull geometry from an existing USGS STAC Item in lieu of the current method that computes the geometry from the `ANG.txt` metadata file. First looks for the USGS STAC in the same location as the XML MTL metadata file. If not there, checks the USGS STAC API. If not found in either location, defaults back to using the `ANG.txt` data for computing the geometry.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x]  Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).
- [ ] Examples are added to `examples/`

pjhartzell committed:
add usgs stac geometry option

pjhartzell committed:
add usgs stac geometry tests

pjhartzell committed:
add some doc text, update command

gadomski commented on tests/test_geometry.py:
```diff
@@ -0,0 +1,71 @@
+
+class GeometryTest(unittest.TestCase):
+
+    def testStoredStac(self):
```
```suggestion
    def test_stored_stac(self):
```

gadomski commented on tests/test_geometry.py:
```diff
@@ -0,0 +1,71 @@
+            self.assertAlmostEqual(e[0], i[0], 7)
+            self.assertAlmostEqual(e[1], i[1], 7)
+
+    def testApiStac(self):
```
```suggestion
    def test_api_stac(self):
```

gadomski commented on tests/test_geometry.py:
```diff
@@ -0,0 +1,71 @@
+            self.assertAlmostEqual(e[0], i[0], 7)
+            self.assertAlmostEqual(e[1], i[1], 7)
+
+    def testAng(self):
```
```suggestion
    def test_ang(self):
```

gadomski reviewed (changes requested):
Since we're using network resources in unit tests, we should probably use [vcrpy](https://vcrpy.readthedocs.io/en/latest/) to record the responses and allow unit tests to run in non-network environments. HMU if you want pointers on setting up vcrpy.

pjhartzell committed:
clean tests function names, add vcr to test_api_stac

- vcr enables the test_api_stac functiont to operate offline even though it hits an "online" resource

pjhartzell commented:
> Since we're using network resources in unit tests, we should probably use [vcrpy](https://vcrpy.readthedocs.io/en/latest/) to record the responses and allow unit tests to run in non-network environments. HMU if you want pointers on setting up vcrpy.

Sweet, yeah, I like this. Implemented.
